docker-compose.yml

```diff
@@ -11,6 +11,9 @@ services:
             - ".rustwide:/home/cratesfyi/rustwide"
         environment:
             CRATESFYI_RUSTWIDE_WORKSPACE: /home/cratesfyi/rustwide
+            CRATESFYI_DATABASE_URL: postgresql://cratesfyi:password@db
+        env_file:
+            - .env
```

Comment:
This prevents people from running `docker-compose up` to get started (they have to run `cp .env.sample .env` first). It looks like [docker-compose automatically reads from .env](https://docs.docker.com/compose/env-file/) even if it's not in the compose file, would that work instead? We could still document it in the README so it doesn't look like magic :)

Reply:
That's a good point, but I'm not sure if what you suggested would help.

As far as I understand yes, docker-compose reads the `.env`, but you still need to forward the variables explicitly to the container you want, and docker-compose will error out if they're missing. That would make error messages less clear.

If we want to retain just one line to get up and running, we could create a wrapper script like this:

```bash
#!/bin/sh
if ! [ -f ".env" ]; then
    cp .env.sample .env
fi
docker-compose "$@"
```

And tell people to do:

```
./compose up
```

Let's also cc @QuietMisdreavus to see what she thinks.

Reply:
Hmm ... could we set the variables to a default and have them be overriden by `.env`? We could put that in the entrypoint, something like `CRATESFYI_GITHUB_USERNAME={$CRATESFYI_GITHUB_USERNAME:-}` (there's probably a way to do that in posix syntax).

Reply:
I think it's fine from a user-experience standpoint to require the `cp` step. Generally speaking, "repo setup" steps like this are not that bad IMO, since they're going to be performed once per clone, so you can just set it and forget it. If you miss it, i'm assuming docker-compose will yell about the missing file so you can check the README or remember the copy step there.

Reply:
Ok, that's reasonable. In that case I think this is ready to merge :)

Reply:
and yes, if it's not there you get `ERROR: Couldn't find env file: /path/to/docs.rs/.env`